### PR TITLE
Fixed requestTimeout issues 

### PR DESCRIPTION
**Pull request by shreys7**
Codegens affected : 

- Added Implementation (was present in getOptions, not actually being used) : 
  - restsharp
  - js-xhr
  - ruby
- Added support for indefinite timeout : 
  - golang (also converted from seconds to milliseconds)
  - java-unirest
  - shell-httpie
  - shell-wget
  - swift
- Added as a whole new option : 
  - libcurl

### CODE REVIEW

File: codegens/csharp-restsharp/lib/restsharp.js [hunk 1]
```diff
@@ -17,7 +17,9 @@ function makeSnippet (request, options) {
     var snippet = `var client = new RestClient("${sanitize(request.url.toString())}");\n`,
         isUnSupportedMethod = UNSUPPORTED_METHODS_LIKE_GET.includes(request.method) ||
             UNSUPPORTED_METHODS_LIKE_POST.includes(request.method);
-
+    if (options.requestTimeout) {
```
Comment (abhijitkane) [inline codegens/csharp-restsharp/lib/restsharp.js]: @shreys7 Can you confirm if not adding a timeout means the code waits indefinitely?

File: codegens/csharp-restsharp/test/unit/convert.test.js [hunk 2]
```diff
@@ -179,7 +178,6 @@ describe('csharp restsharp function', function () {
                 options = {
                     indentCount: 1,
                     indentType: 'tab',
-                    requestTimeout: 2000,
```
Comment (abhijitkane) [inline codegens/csharp-restsharp/test/unit/convert.test.js]: why have we removed this?

File: codegens/golang/lib/index.js [hunk 3]
```diff
@@ -168,7 +168,10 @@ module.exports = {
         }
 
         if (timeout > 0) {
-            codeSnippet += `${indent}timeout := time.Duration(${timeout} * time.Second)\n`;
+            codeSnippet += `${indent}timeout := time.Duration(${timeout / 1000} * time.Second)\n`;
+        }
+        else {
+            codeSnippet += `${indent}timeout := time.Duration(0 * time.Second)\n`;
```
Comment (abhijitkane) [inline codegens/golang/lib/index.js]: Does 0 mean indefinite?
What happens if we don't add this line at all?

File: codegens/golang/test/unit/convert.test.js [hunk 4]
```diff
@@ -125,7 +125,7 @@ describe('Golang convert function', function () {
                     options = {
                         indentCount: 1,
                         indentType: 'tab',
-                        requestTimeout: 2000,
+                        requestTimeout: 5000,
```
Comment (abhijitkane) [inline codegens/golang/test/unit/convert.test.js]: why has this changed?

File: codegens/java-unirest/lib/unirest.js [hunk 5]
```diff
@@ -6,19 +6,22 @@ var _ = require('./lodash'),
...
 function makeSnippet (request, indentString, options) {
     var snippet = '',
         urlString = encodeURI(request.url.toString());
 
     if (options.requestTimeout > 0) {
-        snippet += `Unirest.setTimeouts(${options.requestTimeout}, 0);\n`;
+        snippet += `Unirest.setTimeouts(0, ${options.requestTimeout});\n`;
+    }
+    else {
+        snippet += 'Unirest.setTimeouts(0, 0);\n';
```
Comment (abhijitkane) [inline codegens/java-unirest/lib/unirest.js]: isn't the first arg supposed to be the read timeout?

File: codegens/shell-httpie/test/unit/converter.test.js [hunk 6]
```diff
@@ -144,6 +144,31 @@ describe('Shell-Httpie convert function', function () {
       }
     });
   });
+
+  it.only('should add a timeout of 1 hour (3600 seconds) for RequestTimeout set to 0', function () {
```
Comment (abhijitkane) [inline codegens/shell-httpie/test/unit/converter.test.js]: .only?

File: codegens/shell-httpie/test/unit/converter.test.js [hunk 7]
```diff
@@ -144,6 +144,31 @@ describe('Shell-Httpie convert function', function () {
...
+      }),
+      options = {
+        requestTimeout: 0
+      };
+    convert(request, options, function (error, snippet) {
+      if (error) {
+        expect.fail(null, null, error);
+      }
+      expect(snippet).to.be.a('string');
+      expect(snippet).to.include('--timeout 3600');
```
Comment (abhijitkane) [inline codegens/shell-httpie/test/unit/converter.test.js]: what happens if we don't add `timeout 3600`?

File: codegens/csharp-restsharp/lib/restsharp.js (hunk 1 above)
Comment (shreys7) [reply codegens/csharp-restsharp/lib/restsharp.js]: @abhijitkane The default timeout is 100 seconds. Adding the timeout as -1 is considered as infinite timeout. Will make the requrired changes.

File: codegens/csharp-restsharp/test/unit/convert.test.js (hunk 2 above)
Comment (shreys7) [reply codegens/csharp-restsharp/test/unit/convert.test.js]: @abhijitkane Previously it wasn't used in the code at all. After handling the timeout, some requests are getting bailed out. Will put it back by increasing the timeout to 5 seconds.

File: codegens/java-unirest/lib/unirest.js (hunk 5 above)
Comment (shreys7) [reply codegens/java-unirest/lib/unirest.js]:
```
Unirest.setTimeouts(long connectionTimeout, long socketTimeout);
Connection timeout: The timeout until a connection with the server is established (in milliseconds), and the default value is 10000
Socket timeout: The timeout to receive data (in milliseconds), and the default is 60000.
We can set any of these timeouts to zero to disable the timeout.
```

File: codegens/shell-httpie/test/unit/converter.test.js (hunk 6 above)
Comment (shreys7) [reply codegens/shell-httpie/test/unit/converter.test.js]: Will change this.

File: codegens/shell-httpie/test/unit/converter.test.js (hunk 7 above)
Comment (shreys7) [reply codegens/shell-httpie/test/unit/converter.test.js]: default is 30 seconds. We cannot provide an infinite value, hence 3600 seconds = 1 hour is set.

File: codegens/golang/test/unit/convert.test.js (hunk 4 above)
Comment (shreys7) [reply codegens/golang/test/unit/convert.test.js]: Previously it wasn't used in the code properly as it accepts input in seconds and was used directly . After handling the timeout, some requests are getting bailed out. Hence changed to 5 seconds.

File: codegens/golang/lib/index.js (hunk 3 above)
Comment (shreys7) [reply codegens/golang/lib/index.js]: A Timeout of zero means no timeout.
The default is 0. They haven't provided anywhere in the official docs but read it on a github issue specifying to use different defaults. Will remove the else block.

**Pull request merged**